src/components/LDRouteGuard.vue

```diff
@@ -8,14 +8,20 @@ import ldRedirectMixin from '../mixins/ldRedirect';
 export default {
   mixins: [ldRedirectMixin()],
   props: {
-    component: { type: Object, required: true },
+    component: { type: [Object, Promise], required: true },
     requiredFeatureFlag: { type: String, required: true },
     to: { type: [String, Object], required: true },
   },
   computed: {
     show() {
       return this.$ld.ready && this.$ld.flags[this.requiredFeatureFlag];
     },
+    isPromise() {
+      return !!this.component && typeof this.component.then === 'function';
+    },
```

Comment:
Why expose this as a property?

Reply:
To show the intent of what this check is. Not everyone looking at this is going to realize .then implies a promise.

I guess, in that regard, it makes more sense to name it more specifically.